### PR TITLE
{feat} Enable MSI support for windwos daemonset (MA integration)

### PR DESCRIPTION
**Pull request by bragi92**
This feature enables the windows daemonset with MSI integration on a cluster. 

Please use the following image for testing: 
```
helm pull oci://mcr.microsoft.com/azuremonitor/containerinsights/cidev/prometheus-collector --version 6.0.2-kaveesh-windows-ma-integration-01-09-2023-87f245d8
```

and a command similar to 
```
helm upgrade --install {release_name} .\prometheus-collector-6.0.2-kaveesh-windows-ma-integration-01-09-2023-87f245d8.tgz --dependency-update --namespace=kube-system --set scrapeTargets.windowsExporter=true --set scrapeTargets.windowsKubeProxy=false --set mode.advanced=true --set windowsDaemonset=true --set minimalIngestionProfile="true" --set resources.deployment.requests.cpu="300m" --set resources.deployment.requests.memory="1Gi" --set resources.daemonSet.requests.cpu="150m" --set resources.daemonSet.requests.memory="1Gi" --set useMonitoringAccount=true  --set customEnvironment="azurepubliccloud" --set azureResourceId="{aks_cluster_resource_id}" --set azureResourceRegion="{aks_cluster_region}" 
``` 

if you're trying it out on your cluster.

Full instructions for bug bash can be found [here](https://microsoft.sharepoint.com/teams/ContainerInsights/_layouts/15/Doc.aspx?sourcedoc=%7Bdc357060-d7e5-46a6-a599-88ec4fdee1de%7D&action=edit&wd=target%28Prometheus+Metrics.one%7C29807f1f-70be-4902-b7f6-75b2540c23d3%2FWindows+%28MSI+support+%5C%2F+MA+integration%5C%29%7C6174574a-370e-4183-8b9f-71095a891ed4%2F%29&wdorigin=NavigationUrl&share=IgFgcDXc5demRqWZiOxP3uHeAZZ1TIm7joFoSNIBO-k3t-U)

### CODE REVIEW

File: .pipelines/azure-pipeline-build.yml
Comment (vishiy) [inline .pipelines/azure-pipeline-build.yml:338]: kaveesh, because the deployed multi-arch image tag will be the same for both 19 & 22, so we have telemetry on whether the node is 19 or 22 ?

File: AddonArmTemplate/FullAzureMonitorMetricsProfile.json
Comment (vishiy) [inline AddonArmTemplate/FullAzureMonitorMetricsProfile.json:66]: kaveesh, aren't we pointing to this template from our public docs ? If so, after this is merged, any newer downloads will pick up these rules , which we dont want.

File: otelcollector/configmaps/ama-metrics-prometheus-config-node-windows-configmap.yaml
Comment (vishiy) [inline otelcollector/configmaps/ama-metrics-prometheus-config-node-windows-configmap.yaml:32]: shouldn't the name include windows ?

File: otelcollector/deploy/dashboard/kube-proxy/Kubernetes_Proxy.json
Comment (vishiy) [inline otelcollector/deploy/dashboard/kube-proxy/Kubernetes_Proxy.json:16]: why this change ? all out dashboard changes should be in mixins...

File: otelcollector/configmapparser/default-prom-configs/windowsexporterDefaultRsAdvanced.yml
Comment (bragi92) [reply otelcollector/configmapparser/default-prom-configs/windowsexporterDefaultRsAdvanced.yml]: updated

File: otelcollector/configmapparser/default-prom-configs/windowskubeproxyDefaultRsAdvanced.yml
Comment (bragi92) [reply otelcollector/configmapparser/default-prom-configs/windowskubeproxyDefaultRsAdvanced.yml]: removed

File: otelcollector/configmaps/ama-metrics-prometheus-config-node-windows-configmap.yaml
Comment (bragi92) [reply otelcollector/configmaps/ama-metrics-prometheus-config-node-windows-configmap.yaml]: yes, updated. Thanks for the catch!

File: otelcollector/deploy/dashboard/kube-proxy/Kubernetes_Proxy.json
Comment (bragi92) [reply otelcollector/deploy/dashboard/kube-proxy/Kubernetes_Proxy.json:16]: Not sure, @Sohamdg081992 do you remember why you updated this file here?

File: .pipelines/azure-pipeline-build.yml
Comment (bragi92) [reply .pipelines/azure-pipeline-build.yml:337]: have updated telegraf. Its now collecting a windowsVersion variable which tells us if its a 2019 or a 2022 node

File: otelcollector/deploy/chart/prometheus-collector/templates/prometheus-collector-daemonset.yaml
Comment (bragi92) [reply otelcollector/deploy/chart/prometheus-collector/templates/prometheus-collector-daemonset.yaml]: image
updated

File: otelcollector/deploy/dashboard/kube-proxy/Kubernetes_Proxy.json
Comment (bragi92) [reply otelcollector/deploy/dashboard/kube-proxy/Kubernetes_Proxy.json]: I checked the change. We're updating the proxy selector to the following `kubeProxySelector: 'job=~"kube-proxy|kube-proxy-windows"',`.

I have reverted the change in this file. We will make a new mixin release once this PR merges in.

File: otelcollector/build/windows/scripts/main.ps1
Comment (bragi92) [inline otelcollector/build/windows/scripts/main.ps1:299]: Change logic in ME to work with MAC == false and true both (i.e. localcontrolchannel is not set)

File: AddonArmTemplate/FullAzureMonitorMetricsProfile.json
Comment (bragi92) [reply AddonArmTemplate/FullAzureMonitorMetricsProfile.json:66]: remove this for now. Add a task to merge this in when releasing 3p addon

File: otelcollector/build/windows/scripts/main.ps1
Comment (bragi92) [reply otelcollector/build/windows/scripts/main.ps1:299]: Tested Simple mode : Works as expected.

Comment (bragi92) [reply otelcollector/build/windows/scripts/main.ps1:299]: Tested advanced mode : Works as expected (MAC == true)

File: AddonArmTemplate/FullAzureMonitorMetricsProfile.json
Comment (bragi92) [reply AddonArmTemplate/FullAzureMonitorMetricsProfile.json]: https://github.com/Azure/prometheus-collector/pull/366

**Review by vishiy — APPROVED** [review]

**Pull request merged**
